common_crawler/main.py

```diff
@@ -136,20 +140,29 @@ def get_ls_data() -> list[dict] | None:
     remote_results = response.json()
 
     # Check that the results are valid and usable
+    # Return empty list if project is valid but not yet populated
     # remote_results will resolve to a list if the request is valid,
     # otherwise it will be a dict
-    if type(remote_results) == list and "url" not in remote_results[0]["data"]:
-        print("Column 'url' not present in Label Studio project. Exiting...")
-    elif type(remote_results) == dict and remote_results["status_code"] == 401:
-        print("Invalid Label Studio token passed! Exiting...")
-    elif type(remote_results) == dict and remote_results["status_code"] == 404:
-        print("Invalid Label Studio project ID! Exiting...")
+    if isinstance(remote_results, list):
+        if not remote_results:
+            print("No data in Label Studio project.")
+            return []
+        elif "url" not in remote_results[0]["data"]:
+            print("Column 'url' not present in Label Studio project. Exiting...")
+        else:
+            return remote_results
+    elif isinstance(remote_results, dict):
+        if remote_results.get("status_code") == 401:
+            print("Invalid Label Studio token passed! Exiting...")
+        elif remote_results.get("status_code") == 404:
+            print("Invalid Label Studio project ID! Exiting...")
+        else:
+            print("Unexpected error: ", remote_results)
     else:
-        return remote_results
+        print("Unexpected response type.")
 
     return None
```

Comment:
Two things here:
1. Because this involves a rather complex chain of if conditionals, this section would benefit from being extracted to a separate function
2. In this code, you state "Exiting..." at a few locations but don't actually exit until later. You can save some time and increase comprehensibility of the code by using `sys.exit(message)` at these sections, instead of print statements. If you do this, you can also remove the `return None` statement near the bottom, since the only sections which return anything will be lists.